code/processes/dqe.q

```diff
@@ -6,48 +6,83 @@ detailcsv:@[value;`.dqe.detailcsv;first .proc.getconfigfile["dqedetail.csv"]];
+  `.dqe.results insert (idnum;funct;`$"," sv string raze (),4#params;rs[0];rs[1];.z.p;0Np;0b;"";`started;runtype);
+  }
+
+updresultstab:{[runtype;idnum;end;res;des;status;params;proc]                                                   /- general function used to update a check in the results table
+  .lg.o[`updresultstab;"Updating check id ",(string idnum)," in the results table with status ",string status];
+  if[1b=params`comp;proc:params`compresproc];
+  c:count select from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;                            /- obtain count of checks that will be updated
+  if[c;.lg.o[`updresultstab;raze "run check id ",(string idnum)," update in results table with check status ",string status];
+    `.dqe.results set update endtime:end,result:res,descp:enlist des,chkstatus:status,chkruntype:runtype from .dqe.results where id=idnum,procschk=proc,chkstatus=`started];
+  if[idnum in exec id from .dqe.compcounter;delete from `.dqe.compcounter where id=idnum;];
+  params:()!();
   }
 
-failchk:{[runtype;idnum;error;proc]                                                                            /- general fail function, used to fail a check with inputted error message
-  c:count select from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;
-  if[c;.lg.o[`failerr;raze "run check id ",(string idnum)," update in results table with a fail, with ",(string error)]];
-  `.dqe.results set update chkstatus:`failed,output:0b,descp:c#enlist error,chkruntype:runtype from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;
+chkcompare:{[runtype;idnum;params]                                                                              /- function to compare the checks
+  if[not params[`compcount]=.dqe.compcounter[idnum][`counter];:()];                                             /- checks if all async check results have returned
+  .lg.o[`chkcompare;"comparison started with id ",string idnum];
+  a:.dqe.compcounter[idnum][`results]where not .dqe.compcounter[idnum][`procs]=params`compproc;                 /- obtain all the check returns
+  procsforcomp:.dqe.compcounter[idnum][`procs] except params`compproc;
+  b:.dqe.compcounter[idnum][`results]where .dqe.compcounter[idnum][`procs]=params`compproc;                     /- obtain the check to compare the others to
+  if[all  0W=first b;                                                                                           /- if error in compare proc then fail check
+    .dqe.updresultstab[runtype;idnum;.z.p;0b;"error: error on comparison process";`failed;params;`]];
+  errorprocs:.dqe.compcounter[idnum][`procs] where all each 0W=.dqe.compcounter[idnum][`results];
+  if[(count errorprocs)= count .dqe.compcounter[idnum][`results];                                               /- if error in all comparison procs then fail check
```

Comment:
No error if only a few fail?

Reply:
Any processes that send back an error are stored, then when all the others are compared, the results state how many match, don't match or had errors. The overall check is a fail though as not all came back as matching.